src/renderer/system/archives.js

```diff
@@ -0,0 +1,43 @@
+import ms from 'ms';
+import { lockArchive } from '../../shared/actions/archives';
+import { getAllArchives } from '../../shared/selectors';
+
+const __cache = {
+  timer: null
+};
+
+export const setupArchiveActions = store => ({
+  lockAllArchives() {
+    const archives = getAllArchives(store.getState());
+    if (archives) {
+      archives.forEach(
+        ({ id, status }) =>
+          status === 'unlocked' ? store.dispatch(lockArchive(id)) : ''
+      );
+    }
+  },
+  lockArchiveTimer() {
+    if (__cache.timer) {
+      clearTimeout(__cache.timer);
+    }
+    const state = store.getState();
+
+    const {
+      autolockSeconds,
+      lockArchiveOnFocusout,
+      isButtercupFocused
+    } = state.settings;
+
+    if (
+      (autolockSeconds === '0' &&
+        (lockArchiveOnFocusout && !isButtercupFocused)) ||
+      (autolockSeconds !== '0' &&
+        ((lockArchiveOnFocusout && !isButtercupFocused) ||
+          !lockArchiveOnFocusout))
```

Comment:
This logic should check to see if there is any editing ongoing. If there is, this should fail so no timer is started.

Reply:
I'll try to solve all suggestions in #883

Reply:
Added some checks